crates/wasi-http/wit/deps/http/types.wit

```diff
@@ -33,7 +33,8 @@ interface types {
     invalid-url(string),
     timeout-error(string),
     protocol-error(string),
-    unexpected-error(string)
+    unexpected-error(string),
+    header-name-error(string),
```

Comment:
I think in the case of header errors, the set of failure reasons is disjoint from all the other places that produce `error`s, which I think makes it worth having a separate variant just for header errors such as:
```wit
variant header-error {
  invalid-syntax,
  forbidden
}
```
(I know we don't usually maximally-specialize error variants, but it seems like in cases where they're fully disjoint like this, it's worthwhile.)

Reply:
Note also that if we expand the WASI HTTP spec to distinguish "definitely-forbidden" headers (listed by the WASI HTTP spec) and "host-forbidden" headers (that's purely up to the current running host), we'd probably split `forbidden` into two cases to reflect the difference.

Reply:
I like that specific error type more than extending the more general `error` variant, thanks! I'm not sure about differentiating between definitely forbidden and host forbidden though: what would a guest do differently if it knew the difference between definitely forbidden and host forbidden?